testsuites/syncgateway/upgrade/testserver/sg_replicate/test_upgrade_with_sgreplicate.py

```diff
@@ -0,0 +1,611 @@
+
+    with ProcessPoolExecutor() as up:
+        # Start updates in background process
+        updates_future = up.submit(update_docs, db, cbl_db1, doc_ids,
+                                   cbl_db2, doc_ids2, doc_obj, terminator1_doc_id)
+
+        # 4. Upgrade SGW one by one on cluster config list
+        cluster_util = ClusterKeywords(cluster_config)
+        topology = cluster_util.get_cluster_topology(cluster_config, lb_enable=False)
+        sync_gateways = topology["sync_gateways"]
+        sgw_cluster1_list = sync_gateways[:2]
+        sgw_cluster2_list = sync_gateways[2:]
+        upgrade_sync_gateway(
+            sgw_cluster1_list,
+            sync_gateway_version,
+            sync_gateway_upgraded_version,
+            sgw_cluster1_config_path,
+            cluster_config
+        )
+
+        upgrade_sync_gateway(
+            sgw_cluster2_list,
+            sync_gateway_version,
+            sync_gateway_upgraded_version,
+            sgw_cluster2_config_path,
+            cluster_config
+        )
+
+        cluster = Cluster(config=cluster_config)
+
+        if len(cluster.servers) < 1:
+            raise Exception("Please provide at least 2 servers")
+
+        server_urls = []
+        for server in cluster.servers:
+            server_urls.append(server.url)
+
+        primary_server = cluster.servers[0]
+
+        # 6. Restart SGWs after the server upgrade
+        sg_obj = SyncGateway()
+        for sg in sync_gateways:
+            sg_ip = host_for_url(sg["admin"])
+            log_info("Restarting sync gateway after server upgrade {}".format(sg_ip))
+            sg_obj.restart_sync_gateways(cluster_config=cluster_config, url=sg_ip)
+            time.sleep(5)
+
+        if need_to_redeploy:
+            # Enable xattrs on all SG/SGAccel nodes
+            # cc - Start 1 SG with import enabled, all with XATTRs enabled
+            #    - Do not enable import in SG.
+            if mode == "cc":
+                enable_import = True
+            sg_obj = SyncGateway()
+            for sg in sgw_cluster1_list:
+                sg_ip = host_for_url(sg["admin"])
+                sg_obj.redeploy_sync_gateway_config(
+                    cluster_config=cluster_config,
+                    sg_conf=sgw_cluster1_config_path,
+                    url=sg_ip,
+                    sync_gateway_version=sync_gateway_upgraded_version,
+                    enable_import=enable_import
+                )
+
+            for sg in sgw_cluster2_list:
+                sg_ip = host_for_url(sg["admin"])
+                sg_obj.redeploy_sync_gateway_config(
+                    cluster_config=cluster_config,
+                    sg_conf=sgw_cluster2_config_path,
+                    url=sg_ip,
+                    sync_gateway_version=sync_gateway_upgraded_version,
+                    enable_import=enable_import
+                )
+                # Check Import showing up on all nodes
```

Comment:
is this comment belongs to below line or for loop ?

Reply:
removed this line